x-pack/filebeat/input/azureeventhub/input.go

```diff
@@ -167,6 +167,12 @@ func (a *azureInput) processEvents(event *eventhub.Event, partitionID string) bo
 func (a *azureInput) parseMultipleMessages(bMessage []byte) []string {
 	var mapObject map[string][]interface{}
 	var messages []string
+	// clean up the message for known issues producing a malformed JSON
+	if a.config.SanitizeOptions != nil {
```

Comment:
Nit: A `[]string` will never be nil. It will be an empty slice., over which you can loop (it won't do anything, since it's empty). So this condition isn't required.

Reply:
I agree that the condition serves no purpose. However, can you please clarify on:

> A []string will never be nil. It will be an empty slice.

If you run the following code, Go will actually output `true`:
```
type X struct {
	A []string
}

func main() {
	x := &X{}
	fmt.Println(x.A == nil)
}
```

Reply:
Yes, but it's a `[]string(nil)`.
See https://go.dev/play/p/mBOTE_L2s6E